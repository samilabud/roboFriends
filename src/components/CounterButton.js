import React, {Component} from 'react';

class CounterButton extends Component {
  constructor(props) {
    super(props);
    this.state = {
      count: 0
    };
  }

  shouldComponentUpdate(nextProps, nextState) {
    if (this.state.count !== nextState.count) {
      return true;
    }
    return false;
  }

  render() {
    return (
      <>
      <button
        id='counter'
        color={this.props.color}
        onClick={() => this.setState(state => ({count: state.count + 1}))}>
        Increase Counter
      </button>
      <h1>{this.state.count}</h1>
      </>
    );
  }
}

export default CounterButton