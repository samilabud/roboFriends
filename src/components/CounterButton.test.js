//import {shallow, mount, render} from  'enzyme';//Obsolete
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { create } from "react-test-renderer";
import CounterButton from  './CounterButton';




it('Expect to render CounterButton Component',()=>{
  //expect(shallow(<Card />)).toMatchSnapshot();
  const mockColor = 'red'
  const app = create(<CounterButton color={mockColor} />);
  expect(app.toJSON()).toMatchSnapshot(); 
})


it('Expect to correctly increment of the counter', () => {
  render(<CounterButton />);

  const btnIncrement = screen.getByText("Increase Counter");
  fireEvent.click(btnIncrement);

  expect(screen.getByRole("heading")).toHaveTextContent("1");
});
