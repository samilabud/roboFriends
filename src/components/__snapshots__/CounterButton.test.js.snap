// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render CounterButton Component 1`] = `
Array [
  <button
    color="red"
    id="counter"
    onClick={[Function]}
  >
    Increase Counter
  </button>,
  <h1>
    0
  </h1>,
]
`;
